sound/soc/sof/pm.c

```diff
@@ -136,6 +136,9 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 		if (!sof_ops(sdev)->suspend)
 			return 0;
 
+	/* remove pipelines */
+	sof_destroy_pipelines(dev);
+
```

Comment:
i still think this is controversial.

For system suspend-resume, this is not needed. The DSP will lose context and be powered off.

For pm_runtime suspend-resume, this is needed only on Baytrail.

It feels this should really be done conditionally to avoid doing unnecessary things on all platforms where this is useless.